src/JustSaying/Extensions/TypeExtensions.cs

```diff
@@ -1,20 +1,74 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Reflection;
 using System.Text.RegularExpressions;
 
 namespace JustSaying.Extensions
 {
     public static class TypeExtensions
     {
-        private const int MAX_TOPIC_NAME_LENGTH = 256;
+        private const int MaxTopicNameLength = 256;
+        private const int MaxQueueNameLength = 80;
 
-        public static string ToTopicName(this Type type)
+        private static readonly Dictionary<Type, string> TypeToFriendlyName = new Dictionary<Type, string>
         {
-            var name = type.GetTypeInfo().IsGenericType
-                ? Regex.Replace(type.FullName, "\\W", "_").ToLowerInvariant()
-                : type.Name.ToLowerInvariant();
+            {typeof(string), "string"},
```

Comment:
Could we use the new `[typeof(string)] = "string",` syntax instead?

Reply:
Changed to `HashSet<Type>` rather than a dictionary as it was only storing the lower case name against the type anyway